sdk/servicebus/Azure.Messaging.ServiceBus/src/Receiver/ServiceBusReceiverClient.cs

```diff
@@ -0,0 +1,1296 @@
+        /// <param name="clientOptions">A set of options to apply when configuring the consumer.</param>
+        ///
+        public ServiceBusReceiverClient(
+            string fullyQualifiedNamespace,
+            string topicName,
+            string subscriptionName,
+            TokenCredential credential,
+            ServiceBusReceiverClientOptions clientOptions = default)
+            : this(new ServiceBusConnection(fullyQualifiedNamespace, $"{topicName}/{subscriptionName}", credential, clientOptions?.ConnectionOptions), clientOptions?.Clone() ?? new ServiceBusReceiverClientOptions())
+        {
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="ServiceBusReceiverClient"/> class.
+        /// </summary>
+        ///
+        /// <param name="connection">The <see cref="ServiceBusConnection" /> connection to use for communication with the Service Bus service.</param>
+        /// <param name="clientOptions">A set of options to apply when configuring the consumer.</param>
+        ///
+        public ServiceBusReceiverClient(
+            ServiceBusConnection connection,
+            ServiceBusReceiverClientOptions clientOptions)
+        {
+            Argument.AssertNotNull(connection, nameof(connection));
+            Argument.AssertNotNull(clientOptions, nameof(clientOptions));
+
+            IsSessionReceiver = clientOptions.IsSessionEntity;
+            OwnsConnection = false;
+            Connection = connection;
+            RetryPolicy = clientOptions.RetryOptions.ToRetryPolicy();
+            ReceiveMode = clientOptions.ReceiveMode;
+            Consumer = Connection.CreateTransportConsumer(
+                retryPolicy: RetryPolicy,
+                receiveMode: ReceiveMode,
+                prefetchCount: PrefetchCount,
+                sessionId: clientOptions.SessionId,
+                isSessionReceiver: IsSessionReceiver);
+            Session = new ServiceBusSession(
+                Consumer,
+                clientOptions.SessionId);
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="ServiceBusReceiverClient"/> class.
+        /// </summary>
+        ///
+        protected ServiceBusReceiverClient()
+        {
+            OwnsConnection = false;
+        }
+
+        /// <summary>
+        ///  Receives a batch of <see cref="ServiceBusMessage" /> from the entity using <see cref="ReceiveMode"/> mode.
+        /// </summary>
+        /// <param name="maxMessages">The maximum number of messages that will be received.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        /// <returns></returns>
+        public virtual async IAsyncEnumerable<ServiceBusMessage> ReceiveBatchAsync(
```

Comment:
This is an interesting point. We are actually just returning the the max messages one by one. It might be better to have this just return a Task<IEnumerable> to avoid confusion as to what is happening.
If we expose a truly iterable version, that can return an IAsyncEnumerable<ServiceBusMessage> and we wouldn't include batch in the name.
@ShivangiReja @pakrym thoughts on this?

Reply:
Ended up switching to use `Task<IEnumerable<Message>>`